src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -1030,6 +1034,7 @@ void doStatusCheck()
                         }
                         removeEndpoint(endpoint); // will put it in justRemovedEndpoints to respect quarantine delay
                         evictFromMembership(endpoint); // can get rid of the state immediately
+                        Nodes.peers().remove(endpoint, true, true);
```

Comment:
would you ever call removeEndpoint without also calling Nodes.peers().remove?  should this call just go inside there rather than being placed just after it everywhere?

Reply:
This is quite convoluted actually... 
Given the Cassandra logic, `removeEndpoint` is meant to remove a peer, but it is not meant to remove it from the list of remembered endpoint states. Only `evictFromMembership` is mean to remove the endpoint from completely. Perhaps I could move it there

Reply:
ok.  Just trying to see if we can have less places where the code needs to have added calls to "Nodes" as there seemed to be a few places where the same sets of functions are called, so consolidating them into one place will mean less chance someone adds something in the future and forgets to update everything.